Make block signal driven

mattdodge opened the pull request:
Also separates `WALK` and `GET` commands

hansmosh commented on snmp_base_block.py:
```diff
@@ -0,0 +1,118 @@
+        """ Override this in the child block to make the proper request """
+        raise NotImplementedError()
+
+    def _handle_data(self, varBinds, starting_signal):
```
I see that varBinds is to stick with what pysnmp looks like but what do you think about switching that to var_binds?

hansmosh commented on snmp_base_block.py:
```diff
@@ -0,0 +1,118 @@
+        """
+        raise NotImplementedError()
+
+    def make_snmp_request(self, oids):
```
Being a little picky here, but why not make make_snmp_request _make_snmp_request?

hansmosh commented on snmp_base_block.py:
```diff
@@ -0,0 +1,118 @@
+    exclude_existing = BoolProperty(
+        title="Exclude Existing Values", default=False)
+    lookup_names = BoolProperty(title="Look up OId names", default=False)
+    lookup_values = BoolProperty(title="Look up OId values", default=False)
```
Shouldn't the 'd' be capital here? OID vs OId.

hansmosh commented:
@mattdodge A couple very minor comments on the block for you to review.

mattdodge commented:
Added some unit tests, and did a little refactor. Also addressed review comments. Take a look @hansmosh and let me know if everything still looks good
